compiler/rustc_codegen_gcc/src/common.rs

```diff
@@ -303,6 +303,8 @@ impl<'gcc, 'tcx> ConstCodegenMethods for CodegenCx<'gcc, 'tcx> {
                         let init = self.const_data_from_alloc(alloc);
                         self.static_addr_of(init, alloc.inner().align, None)
                     }
+                    // TODO: generate segment of type id hash
+                    GlobalAlloc::Type { .. } => todo!(),
```

Comment:
This should be mostly the same as the LLVM implementation, except the call to `LLVMConstIntToPtr` will be a call to `Builder::inttoptr` and the call to `const_alloc_to_llvm` should be a call to `CodegenCx::const_data_from_alloc`.